tests/utils/get-toggle-id-by-label.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Get the ID of the setting toogle so test can manipulate the toggle using
+ * unsetCheckbox and setCheckbox utilities.
+ *
+ * We're using 'adaptive timeout' here as the id of the toggle is changed on
+ * every render, so we wait a bit for the toggle to finish rerendering, then
+ * check if the node still attached to the document before returning its
+ * ID. If the node is detached, it means that the toggle is rendered, then
+ * we retry by calling this funtion again with increased retry argument. We
+ * will retry maximum 5 times before returning whatever ID we get.
```

Comment:
Mm, I'm not sure I understand the flow exactly. But this seems to me again to be a candidate for a future headache, because:

1. we are relying on arbitrary timeouts (machines and CI jobs can have different performances and something can lead to a failure in one environment because it is slower than the other, and lead to developer frustration trying to understand why) and
2. also returning “whatever ID we get”.

What does it mean to return whatever ID we get? How does it exactly work? Maybe you can post a video or something? It seems to me that if we have retried 5 times and didn't get what we wanted, we shouldn't return, but `throw`, at the very least.

As for the timeout… it might be the only way possible, I'm not sure. But at least we should throw and fail instead of returning whatever we find, I think.

Reply:
Love this! I improved this function in c7d77b555931c8f9ed8b3390955715bcde5d80d4 according to your review